src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -128,24 +130,29 @@ void Bank::evaluate_and_allocate(std::optional<std::string> weights_path) {
                 return;
             }
 
-            // FIXME: ideally we would want to allocate memory sequentially - in order of UID
+            // Need to allocate
+            uids_to_allocate.insert(device_bank.registered_tensors.at(lt));
```

Comment:
this insert happens under parallel_for, so here's clearly a race 🚩🚩🚩

Reply:
No because of `std::unique_lock<std::mutex> guard(device_bank.mutex);` a couple of lines above

Reply:
Done